File: csharp/main/Int8Def.cs
using System;
using System.Collections.Generic;

namespace Deltix.Luminary
{
	public class Int8Def : IConstantScope, ITypeDef<TypeInt8>
	{
		private Int8Def()
		{
			DefinedConstants.Add(new ConstantDef(this, "MIN_VALUE", TypeInt8.Instance, new LiteralInt8(SByte.MinValue)));
			DefinedConstants.Add(new ConstantDef(this, "MAX_VALUE", TypeInt8.Instance, new LiteralInt8(SByte.MaxValue)));
		}

		public static Int8Def Instance = new Int8Def();

		public String Name => "Int8";

		public String FullName => Name;

		public List<ConstantDef> DefinedConstants { get; } = new List<ConstantDef>();

		public TypeInt8 Type => TypeInt8.Instance;

		public FileDef File => null;

		public ITypeScope Parent => null;
	}
}

File: java/src/main/java/deltix/luminary/Int8Def.java
package deltix.luminary;

import java.util.ArrayList;
import java.util.List;

public class Int8Def implements TypeDef<TypeInt8>, ConstantScope {
    private List<ConstantDef> definedConstants;

    private Int8Def() {
        definedConstants = new ArrayList<>();
        definedConstants.add(new ConstantDef(this, "MIN_VALUE", TypeInt8.INSTANCE, new LiteralInt8(Byte.MIN_VALUE)));
        definedConstants.add(new ConstantDef(this, "MAX_VALUE", TypeInt8.INSTANCE, new LiteralInt8(Byte.MAX_VALUE)));
    }

    public static final Int8Def INSTANCE = new Int8Def();

    @Override
    public String getName() {
        return "UInt8";
    }

    @Override
    public String getFullName() {
        return getName();
    }

    @Override
    public List<ConstantDef> getDefinedConstants() {
        return definedConstants;
    }

    @Override
    public TypeInt8 getType() {
        return TypeInt8.INSTANCE;
    }

    @Override
    public FileDef getFile() {
        return null;
    }

    @Override
    public TypeScope getParent() {
        return null;
    }
}
